File: GrpProj1/Blackjack code in C++/Blackjack-master/Hand.cpp
/* 
 * File:   Hand.cpp
 * Author: Kevin Romero
 * Purpose: Implementation of Hand Object for Blackjack
 * Created on July 12, 2017, 5:16 PM
 */

//System Libraries
#include <vector>

//User Libraries
#include "Hand.h"

//Implement Default Constructor
Hand::Hand(){
    
}

//Calculate hand total
int Hand::getHandTotal(){
    vector<Card *>::const_iterator it=m_hand.begin();
    int n=0;
    
    for(it;it!=m_hand.end();++it){
        n+=(*it)->getCardVal((*it)->getFace());
    }
    handTotal = n;
    return handTotal;
}

//print the hand
void Hand::printHand(){
    vector<Card *>::const_iterator it=m_hand.begin();
    for(it;it!=m_hand.end();++it){
        cout<<(*it)->getGivenCard((*it)->getFace(),(*it)->getSuit());
        cout<<" ";
    }
}

//Draw 2 cards to start round
void Hand::initDraw(Card* c, Card* c2){
    m_hand.push_back(c);
    m_hand.push_back(c2);
}

//Add a card to hand
void Hand::addCard(Card *c){
    m_hand.push_back(c);
}

Hand::~Hand() {
   
}

File: GrpProj1/BlackjackJS/Hand.js
/*
	File: Hand.js 
	Author: Kevin Romero
	Class: CSC17B Team Blackjack
	Purpose: Hand object for Blackjack
	Date: 10/16/17
*/

//Create Hand class
function Hand(){
	//member variables
	this.m_hand=[];
	this.handTotal;
	
	};

	
//add card to hand
Hand.prototype.addCard=function(card){
		this.m_hand.push(card);
};

//draw 2 cards to start round
Hand.prototype.initDraw=function(card1,card2){
	this.m_hand.push(card1);
	this.m_hand.push(card2);
};

//print the hand
Hand.prototype.printHand=function(){
	if(this.m_hand.length<=0){
		document.write("<br>HAND EMPTY<br>");
	}else{

		for(var i=0;i<this.m_hand.length;i++){
			document.write(this.m_hand[i].getSetCard()+"<br>");
		}
	}
};

//calculate hand total
Hand.prototype.getHandTotal=function(){
	var n=0;

	for(var i=0;i<this.m_hand.length;i++){
		n+=this.m_hand[i].getCardVal(this.m_hand[i].getFace());
	}
	this.handTotal = n;
	return this.handTotal;
};

//Accessors
Hand.prototype.getNumCards=function(){
	return this.m_hand.length;
};
